### ISSUE TITLE
Possible issue of reproducibility 

### ISSUE THREAD

**Issue by dawe** [leomorelli/scGET#18]
I analyzed published data and compared final matrices with published ones, I noticed a high number of regions for which the current workflow returns a very high count whereas in published data those are at count 1. I suspect it has to do with the duplicate removal, but I’m working on that. For the time being I suggest a hiatus in using the current workflow or stopping at the initial alignment.

**Comment by dawe** [comment]
I made some tests and I can confirm that counting before merging is wrong, due to the duplicate handling. The proper way (which is actually what we did in the main paper) is to merge 8 bam files before marking duplicates and then counting. Unfortunately this requires more memory and time.
While we work to fix this in the snakefile, a workaround is to manually perform merge/counts like this.

prepare a barcode file like this one

```
CGTACTAG        tn5
TCCTGAGC        tn5
TCATGAGC        tn5
CCTGAGAT        tn5
TAAGGCGA        tnH
GCTACGCT        tnH
AGGCTCCG        tnH
CTGCGCAT        tnH
```

then, for each trasposase `$TN` (tn5 and tnH) merge the 8 barcoded bam files:

```
bclist=( `grep ${TN} barcodes.txt | cut -f 1 | paste - - - -` )
samtools merge -cp ${SAMPLE}_${TN}.bam ${SAMPLE}/${SAMPLE}_BC_${bclist[0]}.bam ${SAMPLE}/${SAMPLE}_BC_${bclist[1]}.bam ${SAMPLE}/${SAMPLE}_BC_${bclist[2]}.bam ${SAMPLE}/${SAMPLE}_BC_${bclist[3]}.bam

samtools index ${SAMPLE}_${TN}.bam
```

After that the duplicate removal and CB are assigned

```
swdir=PATH_TO_scATAC_repo
python ${swdir}/bc2rg.py -w ${SAMPLE}/${SAMPLE}_whitelist.tsv -i ${SAMPLE}/${SAMPLE}_R2.fastq.gz -b ${SAMPLE}_${TN}.bam -O -k -G ${SAMPLE}_${TN} | python ${swdir}/cbdedup.py -I -o ${SAMPLE}_${TN}_bcdedup.bam
samtools index ${SAMPLE}_${TN}_bcdedup.bam
```

Finally, given a set of intervals in `$BEDFILE`, the count for tn5 and tnH can be performed

```python /beegfs/scratch/ric.cosr/cittaro.davide/src/scatACC/para_count.py -p $BEDFILE -b  ${SAMPLE}_${TN}_bcdedup.bam -o $OUTFILE -t 8 
```

This will generate two separate files for tn5 and tnH, they could be loaded and merged into a `AnnData`:


```
import anndata as ad
import scanpy as sc

a5 = sc.read("FILE_TN5.h5ad")
ah = sc.read("FILE_TNH.h5ad")

a5.layers['tn5'] = a5.X
a5.layers['tnH'] = ah.X

a5.write("FINALFILE.h5ad")
```

**Comment by dawe** [comment]
Ok, it does have nothing to do with duplicates. After all, the different tn-barcodes are independent tagmentations and it is correct they are deduped separately. It turns out that when we sum multiple matrices at the end of the workflow, the order of cell names is not conserved, so data are basically shuffled.
This should be fixed in 8b46b760bd56512305a8f92efaead3cec9aeb7bd
